torch/ao/sparsity/sparsifier/weight_norm_sparsifier.py

```diff
@@ -0,0 +1,53 @@
+from functools import reduce
+
+import torch
+import torch.nn.functional as F
+
+from .base_sparsifier import BaseSparsifier
+
+def _flat_idx_to_2d(idx, shape):
+    rows = idx // shape[1]
+    cols = idx % shape[1]
+    return rows, cols
+
+class WeightNormSparsifier(BaseSparsifier):
+    def __init__(self,
+                 sparsity_level=0.5, sparsity_pattern=(1, 4),
+                 zeros_per_block=None):
+        if zeros_per_block is None:
+            zeros_per_block = reduce((lambda x, y: x * y), sparsity_pattern)
```

Comment:
Is this restricted to 2D weights?

Reply:
No, the reduce can take care of any dimensions.